src/metabase/lib/aggregation.cljc

```diff
@@ -439,10 +445,14 @@
    (aggregable-columns query -1 aggregation-position))
   ([query :- ::lib.schema/query
     stage-number :- :int
-    _aggregation-position :- [:maybe ::lib.schema.common/int-greater-than-or-equal-to-zero]]
+    aggregation-position :- [:maybe ::lib.schema.common/int-greater-than-or-equal-to-zero]]
    (let [stage (lib.util/query-stage query stage-number)
+         agg-cols (aggregations-metadata query stage-number)
          columns (into (vec (lib.metadata.calculation/visible-columns query stage-number stage))
-                       (aggregations-metadata query stage-number))]
+                       (cond->> agg-cols
+                         aggregation-position (keep-indexed (fn [i a]
+                                                              (when (not= i aggregation-position)
+                                                                a)))))]
```

Comment:
Possibly dumb questions:
- Doesn't this let you reference aggregations that come later in the list?
- Is that a bug, or intended, or ambiguous?

I know the order is relevant for expressions and joins, so I'm a bit surprised if aggregations are more lenient.